Responsive Side Menu

tmiland opened the pull request:
A layout example with a side menu that hides on mobile.

https://purecss.io/layouts/side-menu/

Did a mock-up of a side menu. I kind of miss it, especially on desktop/laptop where some of the buttons are hard to hit (small surface area). 

Not done anything for light/dark mode, just thought I'd throw the idea out here. 

![Screenshot at 16-06-08](https://user-images.githubusercontent.com/1203834/54212231-8798ad00-44da-11e9-8303-c8ec62b4655b.png)

tmiland commented:
Check it out: https://invidiou.sh 🤓

Perflyst commented:
Looks good but it shouldn't be opened by default. You also cannot hide it!

tmiland commented:
> Looks good but it shouldn't be opened by default. You also cannot hide it!

Thanks.

That's default behaviour.

>  /*
> Hides the menu at `48em`, but modify this based on your app's needs.
> */

omarroth commented:
Looks very nice. Unfortunately it appears to have bullet points in Chrome:
![image](https://user-images.githubusercontent.com/14208607/54324135-657e5800-45ca-11e9-909c-60b5c2a28ece.png).

How would you like this PR to be handled? It's definitely a nice idea. I'm currently not sure how well it would be integrated into the rest of the site. Since it has links to subscriptions, trending, etc. would it make sense to remove the feed menu along the top? Also as mentioned, it should be possible to hide it. Using:

```diff
diff --git a/assets/css/side-menu.css b/assets/css/side-menu.css
index 7572752..ca514ac 100644
--- a/assets/css/side-menu.css
+++ b/assets/css/side-menu.css
@@ -207,7 +207,7 @@ small screens.
 /*
 Hides the menu at `48em`, but modify this based on your app's needs.
 */
-@media (min-width: 48em) {
+@media (min-width: 200em) {
 
     .header,
     .content {
```
Makes it hidden by default, although if possible it would make sense to remove the `@media` for the given styling.

omarroth commented on src/invidious/views/template.ecr:
```diff
@@ -28,118 +29,205 @@
+        </li>
+        <li class="pure-menu-item menu-item-divided">
+          <a title="<%= translate(locale, " Preferences") %>" href="/preferences?referer=
+            <%= env.get?("current_page") %>" class="pure-menu-link">
```
This should be `href="/preferences?referer=<%= env.get?("current_page") %>"`, since otherwise the given URL contains the leading spaces, so it links to `/preferences?referer=             %2Ffeed%2Fsubscriptions`. If you want, you can still spread the attributes over multiple lines, something like:
```ecr
<a title="<%= translate(locale, " Preferences") %>"
href="/preferences?referer=<%= env.get?("current_page") %>"
class="pure-menu-link">
```

tmiland replied:
Seems like this was the Atom "beautifier" who broke this. Should be fixed now.

omarroth commented on src/invidious/views/template.ecr:
```diff
@@ -28,118 +29,205 @@
+            <div class="pure-u-1-4">
+              <a href="/signout?referer=<%= env.get?(" current_page") %>&token=
+                <%= env.get?("token") %>&challenge=
+                <%= env.get?("challenge") %>" class="pure-menu-heading">
```
See above.

omarroth commented on src/invidious/views/template.ecr:
```diff
@@ -28,118 +29,205 @@
+            </div>
+            <div class="pure-u-1-4">
+              <a title="<%= translate(locale, " Preferences") %>" href="/preferences?referer=
+                <%= env.get?("current_page") %>" class="pure-menu-heading">
```
See above.

omarroth commented on src/invidious/views/template.ecr:
```diff
@@ -28,118 +29,205 @@
+            </div>
+            <div class="pure-u-1-3">
+              <a title="<%= translate(locale, " Preferences") %>" href="/preferences?referer=
+                <%= env.get?("current_page") %>" class="pure-menu-heading">
```
See above.

tmiland commented:
> Looks very nice. Unfortunately it appears to have bullet points in Chrome:
> ![image](https://user-images.githubusercontent.com/14208607/54324135-657e5800-45ca-11e9-909c-60b5c2a28ece.png).
> 
> How would you like this PR to be handled? It's definitely a nice idea. I'm currently not sure how well it would be integrated into the rest of the site. Since it has links to subscriptions, trending, etc. would it make sense to remove the feed menu along the top? Also as mentioned, it should be possible to hide it. Using:
> 
> ```diff
> diff --git a/assets/css/side-menu.css b/assets/css/side-menu.css
> index 7572752..ca514ac 100644
> --- a/assets/css/side-menu.css
> +++ b/assets/css/side-menu.css
> @@ -207,7 +207,7 @@ small screens.
>  /*
>  Hides the menu at `48em`, but modify this based on your app's needs.
>  */
> -@media (min-width: 48em) {
> +@media (min-width: 200em) {
>  
>      .header,
>      .content {
> ```
> 
> Makes it hidden by default, although if possible it would make sense to remove the `@media` for the given styling.

I'm not exactly sure. The content of the menu is just a mock-up.

My thoughts though are to have the various links gathered in the menu, like youtube. 

Not sure how to handle the above reviews, do i just mark them as resolved, and it will be merged?

I have added the above suggestions to the css, and Removed bullet points in Chrome.

Pushed it to the site just now. 🙂

Perflyst commented:
I wonder if you can design this so that we have a bit more youtube like layout. The videos are currently very big at the front page. If you make the videos a bit smaller you could expand it on the startpage but hide it everywhere else by default.

It also overwrites the `Invidious` on the top. Maybe you can really take a look at youtube.com how they expand and hide it.

Your side menu is also dark design while the startpage is white.

tmiland commented:
> I wonder if you can design this so that we have a bit more youtube like layout. The videos are currently very big at the front page. If you make the videos a bit smaller you could expand it on the startpage but hide it everywhere else by default.
> 
> It also overwrites the `Invidious` on the top. Maybe you can really take a look at youtube.com how they expand and hide it.
> 
> Your side menu is also dark design while the startpage is white.

Thanks for the feedback @Perflyst 

Check out the latest changes. :smile:

dimqua commented:
IMHO, if you're going to add `Preferences` link to the side menu, it should be removed from the top menu (or vice versa).
